apis/v1alpha1/integration.go

```diff
@@ -0,0 +1,151 @@
+	// Specifies how to handle request payload content type conversions. Supported
+	// values are CONVERT_TO_BINARY and CONVERT_TO_TEXT, with the following behaviors:
+	//
+	// If this property is not defined, the request payload will be passed through
+	// from the method request to integration request without modification, provided
+	// that the passthroughBehavior is configured to support payload pass-through.
+	ContentHandling *string `json:"contentHandling,omitempty"`
+	// Specifies whether credentials are required for a put integration.
+	Credentials *string `json:"credentials,omitempty"`
+	// Specifies the HTTP method for the integration.
+	// +kubebuilder:validation:Required
+	HTTPMethod *string `json:"httpMethod"`
+	// The HTTP method for the integration.
+	IntegrationHTTPMethod *string `json:"integrationHTTPMethod,omitempty"`
+	// Specifies the pass-through behavior for incoming requests based on the Content-Type
+	// header in the request, and the available mapping templates specified as the
+	// requestTemplates property on the Integration resource. There are three valid
+	// values: WHEN_NO_MATCH, WHEN_NO_TEMPLATES, and NEVER.
+	PassthroughBehavior *string `json:"passthroughBehavior,omitempty"`
+	// A key-value map specifying request parameters that are passed from the method
+	// request to the back end. The key is an integration request parameter name
+	// and the associated value is a method request parameter value or static value
+	// that must be enclosed within single quotes and pre-encoded as required by
+	// the back end. The method request parameter value must match the pattern of
+	// method.request.{location}.{name}, where location is querystring, path, or
+	// header and name must be a valid and unique method request parameter name.
+	RequestParameters map[string]*string `json:"requestParameters,omitempty"`
+	// Represents a map of Velocity templates that are applied on the request payload
+	// based on the value of the Content-Type header sent by the client. The content
+	// type value is the key in this map, and the template (as a String) is the
+	// value.
+	RequestTemplates map[string]*string `json:"requestTemplates,omitempty"`
+	// Specifies a put integration request's resource ID.
+	ResourceID  *string                                  `json:"resourceID,omitempty"`
+	ResourceRef *ackv1alpha1.AWSResourceReferenceWrapper `json:"resourceRef,omitempty"`
+	// The string identifier of the associated RestApi.
+	RestAPIID  *string                                  `json:"restAPIID,omitempty"`
+	RestAPIRef *ackv1alpha1.AWSResourceReferenceWrapper `json:"restAPIRef,omitempty"`
+	// Custom timeout between 50 and 29,000 milliseconds. The default value is 29,000
+	// milliseconds or 29 seconds.
+	TimeoutInMillis *int64     `json:"timeoutInMillis,omitempty"`
+	TLSConfig       *TLSConfig `json:"tlsConfig,omitempty"`
+	// Specifies a put integration input's type.
+	// +kubebuilder:validation:Required
+	Type *string `json:"type_"`
```

Comment:
You can override this tag to `json:"type"` using something like https://github.com/aws-controllers-k8s/eks-controller/blob/main/generator.yaml#L445-L446

Reply:
Thanks! I was looking for this feature but could not find it in code-generator. I saw another controller using the `type_` field as well.

Reply:
For posterity, this change also ended up removing `type_` from the required field list. It's a bug in code-generator.